functions.php

```diff
@@ -1,55 +1,78 @@
+        ON `films`.`genre` = `genre`.`id` 
+        WHERE `is_deleted` = false' );
     $query->execute();
     $result = $query->fetchAll();
     return $result;
 }
 
-function getAllRolesForFilms(string $db_name):  array
+/**
+ * Function to get all the roles I have done, for each of the films in the collection, where the film is not flagged as being deleted.
+ * The DB contains a table of films, plus a linked table of multiple roles/jobs that I have undertaken on each file
+ */
+function getAllRolesForFilms(object $db):  array
 {
-    $dsn = 'mysql:host=db;dbname=' . $db_name;
-    $db = new PDO($dsn,'root', 'password');
-    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
-    //SQL to find all the roles I did for each film I worked on
-    $query = $db->prepare('select `films`.`id` as `film-id`, `name`, `roles`.`id` from `films` 
-        join `film_roles`
-        on `films`.`id` = `film_roles`.`film_id` 
-        join `roles`
-        on `roles`.`id` = `film_roles`.`role_id`' );
+    $query = $db->prepare('SELECT `films`.`id` AS `film-id`, `name`, `roles`.`id` FROM `films` 
+        JOIN `film_roles`
+        ON `films`.`id` = `film_roles`.`film_id` 
+        JOIN `roles`
+        ON `roles`.`id` = `film_roles`.`role_id` 
+        WHERE `is_deleted` = false' );
     $query->execute();
     $result = $query->fetchAll();
     return $result;
 }
 
-function displayFilmsAndRoles($result_films, $result_roles): string
+/**
+ * Function to display all the films in the collection, and for each film, display all the roles that I did for that film. If film id or title is missing, dont show any details of the film and move onto showing the next film
+ */
+function displayFilmsAndRoles(array $result_films, array $result_roles): string
 {
     $film_results = "";
-    // show all the films
-    foreach($result_films as $key => $value) {
-        $film_results .= '<article class="container__film">';
-        $film_results .= '<h2>Film: ' . $result_films[$key]['title'] . '</h2>' ;
-//      $film_results .= '<p>ID: ' . $result_films[$key]['id'] . '</p>';
-        $film_results .= '<p>Year Produced: ' .  $result_films[$key]['year_produced'] . '</p>';
-        $film_results .= '<p>Genre: ' . $result_films[$key]['type'] . '</p>';
-        $film_results .= '<p>My Roles: </p>';
-        $film_results .= '<ul>';
-        //for each film, shows the roles I did
-        foreach($result_roles as $key_role => $value_role) {
-            if ($result_films[$key]['id'] == $result_roles[$key_role]['film-id']) {
-                $film_results .= '<li>' . $result_roles[$key_role]['name']   . '</li>' ;
+    //iterate over outer array of films, and show each film
+    foreach($result_films as $film) {
+        //check that mandatory fields id (auto-incremented primary key) and title are present & not null...
+        if ( array_key_exists('id', $film)
+            && array_key_exists('title', $film)
+            && ($film['title'] != "") )
```

Comment:
formatting for this should be as follows:
if (
    array_key_exists('id', $film) && 
    array_key_exists('title', $film)&& 
    ($film['title'] != "")
) {

Apparently this reads nicer.